pkg/service/backup/restore_worker_tables.go

```diff
@@ -19,18 +24,41 @@ type tablesWorker struct {
 
 // restoreData restores files from every location specified in restore target.
 func (w *tablesWorker) restore(ctx context.Context, run *RestoreRun, target RestoreTarget) error {
-	w.AwaitSchemaAgreement(ctx, w.clusterSession)
+	if target.Continue && run.PrevID != uuid.Nil && run.Table != "" {
+		w.continuation = true
+	}
 	if !w.continuation {
 		w.initRemainingBytesMetric(ctx, run, target)
 	}
 
+	if run.Stage.Index() <= StageRestoreData.Index() {
+		if err := w.stageRestoreData(ctx, run, target); err != nil {
+			return errors.Wrap(err, "restore data")
+		}
+	}
+
+	if run.Stage.Index() <= StageRestoreRepair.Index() {
+		if err := w.stageRepair(ctx, run, target); err != nil {
+			return errors.Wrap(err, "repair")
+		}
+	}
+
+	return nil
+}
+
+func (w *tablesWorker) stageRestoreData(ctx context.Context, run *RestoreRun, target RestoreTarget) error {
+	run.Stage = StageRestoreData
+	w.insertRun(ctx, run)
+
+	w.AwaitSchemaAgreement(ctx, w.clusterSession)
 	w.Logger.Info(ctx, "Started restoring tables")
 	defer w.Logger.Info(ctx, "Restoring tables finished")
-	// Disable gc_grace_seconds
+
+	// Disable tombstone_gc
 	for _, u := range run.Units {
 		for _, t := range u.Tables {
-			if err := w.DisableTableGGS(ctx, u.Keyspace, t.Table); err != nil {
-				return errors.Wrap(err, "disable table's ggs")
+			if err := w.AlterTableTombstoneGC(ctx, u.Keyspace, t.Table, modeDisabled); err != nil {
+				return errors.Wrap(err, "disable table's tombstone_gc")
 			}
 		}
 	}
```

Comment:
The assumption is that one unit represents tables from one keyspace and that there are no two units of the same keyspace. This can be seen in the `newUnits` implementation.